### PR TITLE
[Merged by Bors] - feat: port NumberTheory.Padics.PadicIntegers

### PR DESCRIPTION
**Pull request by xroblot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/Padics/PadicIntegers.lean [hunk 1]
```diff
@@ -0,0 +1,703 @@
...
+#align padic_int.norm_lt_one_add PadicInt.norm_lt_one_add
+
+theorem norm_lt_one_mul {z1 z2 : ℤ_[p]} (hz2 : ‖z2‖ < 1) : ‖z1 * z2‖ < 1 :=
+  calc
+    ‖z1 * z2‖ = ‖z1‖ * ‖z2‖ := by simp
+    _ < 1 := mul_lt_one_of_nonneg_of_lt_one_right (norm_le_one _) (norm_nonneg _) hz2
+
+#align padic_int.norm_lt_one_mul PadicInt.norm_lt_one_mul
+
+@[simp, nolint simpNF] -- Porting note: simpNF wants to change the LHS to ¬ IsUnit z
```
Comment (ChrisHughes24) [inline Mathlib/NumberTheory/Padics/PadicIntegers.lean:465]: The linter is correct to flag this lemma. A better solution than switching off the linter should be found, maybe adding other lemmas, or maybe just removing `simp` from this lemma altogether.

File: Mathlib/NumberTheory/Padics/PadicIntegers.lean [hunk 2]
```diff
@@ -0,0 +1,703 @@
...
+#align padic_int.inv PadicInt.inv
+
+instance : CharZero ℤ_[p]
+    where cast_injective m n h :=
+    Nat.cast_injective (
+      show (m : ℚ_[p]) = n by
+        rw [Subtype.ext_iff] at h
+        norm_cast at h)
+
+@[simp 1001, norm_cast] -- Porting note: simpNF linter
```
[the remark this answers is not in the document]
Comment (xroblot) [reply Mathlib/NumberTheory/Padics/PadicIntegers.lean]: Removed.

File: Mathlib/NumberTheory/Padics/PadicIntegers.lean [hunk 3]
```diff
@@ -0,0 +1,703 @@
...
+
+@[simp]
+theorem norm_eq_padic_norm {q : ℚ_[p]} (hq : ‖q‖ ≤ 1) : @norm ℤ_[p] _ ⟨q, hq⟩ = ‖q‖ := rfl
+#align padic_int.norm_eq_padic_norm PadicInt.norm_eq_padic_norm
+
+@[simp]
+theorem norm_p : ‖(p : ℤ_[p])‖ = (p : ℝ)⁻¹ := padicNormE.norm_p
+#align padic_int.norm_p PadicInt.norm_p
+
+@[simp 1001] -- Porting note: simpNF linter
```
[the remark this answers is not in the document]
Comment (xroblot) [reply Mathlib/NumberTheory/Padics/PadicIntegers.lean]: Removed.

File: Mathlib/NumberTheory/Padics/PadicIntegers.lean (hunk 1 above)
Comment (xroblot) [reply Mathlib/NumberTheory/Padics/PadicIntegers.lean]: I guess the easiest solution is to remove the `simp` so I removed this one too ;)

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**